3.4 homework

Likhachev17 opened the pull request.

Likhachev17 committed:
-Удалены комментарии из pubspec.yaml, созданные при генерации проекта
-Подключена библиотека flutter_svg
-Хранилищем ассетов была установлена папка res
-Добавлен шрифт Roboto

Likhachev17 committed:
-Удалены комментарии из pubspec.yaml, созданные при генерации проекта
-Подключена библиотека flutter_svg
-Хранилищем ассетов была установлена папка res
-Добавлен шрифт Roboto

Jokerkras commented on local.properties:
```diff
@@ -0,0 +1,8 @@
+## This file must *NOT* be checked into Version Control Systems,
```
В одной из домашек должен был быть пример gitignore, скопируйте его от туда или хотя бы добавьте туда этот файл

Likhachev17 replied:
Здравствуйте! В корне проекта лежит gitignore с текстом взятым в домашке, я что-то напутал?

Jokerkras replied:
добавьте тогда просто два файла, к которым я комментарий написал в gitignore. я еще раз гляну все ли присутсвует в уроке с gitignore

Jokerkras commented on pubspec.lock:
```diff
@@ -43,13 +43,34 @@ packages:
       url: "https://pub.dartlang.org"
```
В одной из домашек должен был быть пример gitignore, скопируйте его от туда или хотя бы добавьте туда этот файл

Jokerkras commented on pubspec.lock:
```diff
@@ -62,6 +83,13 @@ packages:
+      name: flutter_svg
+      url: "https://pub.dartlang.org"
+    source: hosted
+    version: "0.14.4"
```
Как можете заметить при использовании ^0.14.0 у вас максимум подтянулась версия 0.14.4 - это последняя из 0.14, можете почитать подробней про символ ^ в pubspec

Jokerkras commented on pubspec.yaml:
```diff
@@ -24,53 +11,34 @@ dependencies:
   cupertino_icons: ^1.0.0
 
+  dependencies:
+  flutter_svg: ^0.14.0
```
Вы просто решили использовать версию пониже, или просто нашли пример? Для использования актуальных версий можете зайти сразу на https://pub.dev/packages/flutter_svg/install, тут обычно уже минимальная информация, необходимая для использования и подключения

Likhachev17 replied:
Версия ^0.19.0 Завелась При попытке поставить актуальную 0.19.2+1 была ругань:

The current Flutter SDK version is 1.22.5.

Because places depends on flutter_svg >=0.19.2 which requires Flutter SDK version >=1.24.0-6.0.pre, version solving failed.
pub get failed (1; Because places depends on flutter_svg >=0.19.2 which requires Flutter SDK version >=1.24.0-6.0.pre, version solving failed.)

Likhachev17 committed:
-Исправлена версия flutter svg с ^0.14.0 на ^0.19.0
-Был дополнен файл gitignore в соответствии с документом приложенным к заданию 3.2

Likhachev17 committed:
-Исправлена версия flutter svg с ^0.14.0 на ^0.19.0
-Был дополнен файл gitignore в соответствии с документом приложенным к заданию 3.2

Likhachev17 committed:
Запустите приложение. С помощью hot reload инициируйте перерисовку.
Чему равно значение счетчика? Почему значение именно такое?
Значение счетчика увеличилось на единицу, потому-что hot reload предназначен для обновления приложения на основе текущего состояния,
в то время как hot restart полностью перезапускает приложение восстанавливая исходное состояние.

В случае с Stateless значение счетчика всегда будет 1, т.к. виджет создается заново.
В случае с Statefull значение будет каждый раз увеличиваться на 1, т.к. виджет перерисовывается на основе текущего состояния.

pretorean commented on .gitignore:
```diff
@@ -30,6 +30,8 @@
 .pub/
 /build/
+local.properties
+pubspec.lock
```
теперь эти файлики надо удалить и сделать коммит, потом сделать
```
flutter clean
flutter pub get
```
и тогда изменения в этих файлах не будут попадать в репо
